### ISSUE TITLE
Updated analysis: Rerun all molecular subtyping modules with consensus+hotspot SNV MAF

### ISSUE THREAD

**Issue by jharenza** [AlexsLemonade/OpenPBTA-analysis#959]
#### What analysis module should be updated and why?
Once #819 is complete, rerun molecular subtyping modules, as some subtypes previously missed will now be detected. 

#### What changes need to be made? Please provide enough detail for another participant to make the update.
- rerun all molecular subtyping modules
- some samples (PNOC) otherwise missed might now be subtyped with the consensus + hotspot SNV MAF. Those were previously added via clinical subtyping updates, so we could either keep both and deduplicate or possibly archive the clinical subtyping piece.

#### What input data should be used? Which data were used in the version being updated?
consensus+hotspot MAF

#### When do you expect the revised analysis will be completed?
post review of all PRs for #819 

#### Who will complete the updated analysis?
@kgaonkar6

**Comment by kgaonkar6** [comment]
@jharenza  @jaclyn-taroni 
I think we would need some order in which PR go in to get this updated, how does the following sound (open to suggestions ):

1.  Update code only in subtyping modules (CRANIO, LGAT and HGG) requiring snv maf files:
```
file.path(root_dir,
                                        "data",
                                        "pbta-snv-consensus-mutation.maf.tsv.gz")
```
to 
```
file.path(root_dir,
                              "analyses" , "hotspots-detection" , "results" , "pbta-snv-scavenged-hotspots.maf.tsv.gz")
```

2. Rerun preprocessing modules used in subtyping with relative path to hotspot mafs and updated cnv consensus files (tp53-nf1-score needs update to use hotspots maf as well an intermediate file from `focal-cn-file-preparation`)

 3. Rerun subtyping with the updated files from 1 and 2 and summarise changes 
 4. If subtypes are missed by using hotspot maf only, combine with consensus maf file and deduplicate the calls.
 5. Review the changes in 4.

**Comment by jharenza** [comment]
@kgaonkar6 I think this sounds ok but more time consuming than perhaps taking the modules one by one. Since we look for mutations in an exon (CRANIO) and kinase domains (LGAT), even if the hotspots maf is sufficient for these (I am guessing it will miss some mutations), then I would still like to see the consensus + hotspot mafs in these subtyping modules so that we still capture all possible SNVs when future samples are added. For HGG, it looks like from `H3F3A K28M, H3F3A G35R/V, HIST1H3B K28M, HIST1H3C K28M, HIST2H3C K28M`, the hotspots file does not contain HIST1H3C K28M or HIST2H3C K28M, which would also warrant using the consensus + hotspot maf. In addition, while we have two _TERT_ promoter hotspots in the hotspot maf, we will be missing any other mutations if we do not use the consensus maf and which we observe in the [SNV results file](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-HGG/results/HGG_cleaned_mutation.tsv). So, in the interest of time, I think it is fine to just go ahead with the consensus+maf merge for these modules from the get-go.

**Comment by jaclyn-taroni** [comment]
A couple comments on what has been discussed so far:

* I agree with taking the modules one by one. I'm not convinced we want code only PRs; we are certainly concerned with how the changes impact with the results! So, going through each module and submitting a PR that also includes a rerun and the results seems like a good way to approach this. (This is to say that I think we should close #1071.)
* It seems reasonable to use a consensus and hotspot merge for these subtyping modules and very clearly document that choice given the _goal of these modules._ 
* To me, it makes sense to _start_ with the modules that are upstream of any subtyping as the first pull requests.

**Comment by kgaonkar6** [comment]
Ok sound good, closed #1071 and will update subtyping only modules with consensus + hotspot maf

**Comment by kgaonkar6** [comment]
Closing this issue with #1075, #1076 and #1077
